### PR TITLE
test: ensure only `sdk.DefaultBondDenom` is used for fees in our modules simulation

### LINKED ISSUE #369: Simulation test: `out of gas` error for some simulation
**Describe the bug**
When setting values for `FlagNumBlocksValue` and `FlagBlockSizeValue` flags, the simulation test can fail with the error `out of gas in location: WritePerByte; gasWanted: 1000000, gasUsed: 1000373: out of gas`
While it particularly happen when setting high values, the error can sometimes occurs for random low values. Therefore, there might be a flaw in our configuration for simulation tests 

**To Reproduce**
Set following values in `app/simulation_test.go`
```
simapp.FlagNumBlocksValue = 500
simapp.FlagBlockSizeValue = 100
```
```
Simulating... block 50/500, operation 0/273.    simulate.go:295: error on block  50/500, operation (14/80) from x/staking:
        out of gas in location: WritePerByte; gasWanted: 1000000, gasUsed: 1000373: out of gas
```

### PR DESCRIPTION
**Pull request by giunatale**
Closes #369 

This PR does not really fix #369, as explained in the issue itself.
However, we at least switch over using only the `sdk.DefaultBondDenom` coin for fees in our modules' simulation.
Problem will stil exist however for modules we pull from the cosmos-sdk, so `out of gas` will still occasionally occur (and it's highly likely to occur for large simulations), but I propose we close #369 as this can't really be avoided.

I've also made sure we initialize the `rand` seed in `TestAppStateDeterminism`, and fixed an error in how we were selecting auctions for `msgParticipate` simulation. 

Apologies if I may have bundled too many things together in this PR.

### CODE REVIEW

**Review by aljo242 — APPROVED** [review]
Do we need `RandomAuctionStandby` at this point?

**Comment by giunatale** [comment]
> Do we need `RandomAuctionStandby` at this point?

Yes, for `msgCancelAuction`

File: app/simulation_test.go [hunk 1]
```diff
@@ -118,6 +118,7 @@ func TestAppStateDeterminism(t *testing.T) {
 	config.OnOperation = true
 	config.AllInvariants = true
 
+	rand.Seed(time.Now().Unix())
```
Comment (lubtd) [inline app/simulation_test.go:121]: Is it for the simulation seed? Can't we have a flag for it?

Comment (giunatale) [reply app/simulation_test.go:121]: we use `rand.Int63()` below

We could use the seed flag, but why? We are not seeking repeatability, but rather find non-determinism. @aljo242 opinions?

I could use the `rand.Seed(simapp.FlagSeedValue)` in any case, if wanted

**Review by lubtd — APPROVED** [review]

**Pull request merged**